src/main.ts

```diff
@@ -21,4 +49,133 @@ async function run() {
   }
 }
 
+async function handleMain(inputs: Inputs, client: GitHubClient): Promise<void> {
+  const changes = await getChangesInPush();
+
+  if (changes.additions.includes(inputs.haltFile)) {
+    core.startGroup(`${inputs.defaultBranch}:${inputs.haltFile} added`);
+    const message = haltMessage(fs.readFileSync(inputs.haltFile).toString());
+    core.info(`Halting all open PRs: ${message}`);
+    await haltOpenPullRequests(client, github.context, inputs, message);
+    core.endGroup();
+  }
+
+  if (changes.removals.includes(inputs.haltFile)) {
+    core.startGroup(`${inputs.defaultBranch}:${inputs.haltFile} removed`);
+    core.info("Un-halting all open PRs");
+    await unhaltOpenPullRequests(client, github.context, inputs);
+    core.endGroup();
+  }
+}
+
+async function handlePullRequest(
+  inputs: Inputs,
+  client: GitHubClient,
+  pullRequest: PullRequest
+): Promise<void> {
+  const haltFile = await githubApi.getRepositoryContent(client, {
+    ...github.context.repo,
+    path: inputs.haltFile,
+    ref: inputs.defaultBranch,
+  });
+
+  const haltFileContents =
+    haltFile && "content" in haltFile ? atob(haltFile.content) : null;
+
+  if (!haltFileContents) {
+    core.info("Repository not halted");
+    return await unhaltPullRequest(client, github.context, inputs, pullRequest);
+  }
+
+  const changes = await getChangesInPullRequest(
+    client,
+    github.context,
+    pullRequest
+  );
+
+  if (changes.removals.includes(inputs.haltFile)) {
+    core.info(`${inputs.defaultBranch}:${inputs.haltFile} exists, but removed`);
+    return await unhaltPullRequest(client, github.context, inputs, pullRequest);
+  }
```

Comment:
So any PR with a removed file will unlock the halting? I have a feeling there are some race type conditions laying around in this behavior where `main` and the PR can get out of sync and the original halt removed in the PR is no longer the current halt on `main`.

Reply:
Hmm, I can see the spidey-sense that this _could_ be an issue, but I couldn't come up with any actual concrete scenarios. Can you come up with a hypothetical timeline of a Bad Outcome?